presto-main/src/main/java/com/facebook/presto/operator/scalar/UrlFunctions.java

```diff
@@ -129,6 +132,42 @@ public static Slice urlExtractParameter(@SqlType(StandardTypes.VARCHAR) Slice ur
         return null;
     }
 
+    @Nullable
+    @Description("encode url")
+    @ScalarFunction
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice urlEncode(@SqlType(StandardTypes.VARCHAR) Slice url)
+    {
+        if (url == null) {
+            return null;
+        }
+
+        try {
+            return slice(URLEncoder.encode(url.toString(UTF_8), UTF_8.name()));
+        }
+        catch (UnsupportedEncodingException e) {
+            return null;
+        }
+    }
+
+    @Nullable
+    @Description("decode url")
+    @ScalarFunction
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice urlDecode(@SqlType(StandardTypes.VARCHAR) Slice url)
+    {
+        if (url == null) {
+            return null;
+        }
+
+        try {
+            return slice(URLDecoder.decode(url.toString(UTF_8), UTF_8.name()));
+        }
+        catch (UnsupportedEncodingException e) {
+            return null;
```

Comment:
This should throw an AssertionError. If UTF-8 encoding is not available in the JVM, something is very wrong.

Unfortunately, there's no equivalent functionality in Guava for the decoder side. So until this is fixed (https://github.com/google/guava/issues/1691) we're going to have to continue to use URLDecoder here.